ui/server/src/main/scala/pl/touk/nussknacker/ui/api/ManagementResources.scala

```diff
@@ -122,11 +145,18 @@ class ManagementResources(processCounter: ProcessCounter,
   private implicit final val plainBytes: FromEntityUnmarshaller[Array[Byte]] = Unmarshaller.byteArrayUnmarshaller
   private implicit final val plainString: FromEntityUnmarshaller[String] = Unmarshaller.stringUnmarshaller
 
-  private def withComment: Directive1[Option[String]] =
+  //if all deployment comments parsed come through here isn't it enough to validate that method?
+  private def withDeploymentComment: Directive1[Option[DeploymentComment]] = {
     entity(as[Option[String]]).map(_.filterNot(_.isEmpty)).flatMap {
       case None if deploySettings.exists(_.validationPattern.nonEmpty) => reject(ValidationRejection("Comment is required", None))
-      case comment => provide(comment)
+      case Some(comment) =>
+        DeploymentComment(comment, deploySettings.get) match {
```

Comment:
I think method apply should looks like:
```
def apply(comment: String, deploySettings: Option[DeploySettings]) 
```
After that you can put just deploySettings.

Reply:
changed it and used in validating as well, thanks